src/java.xml/share/classes/org/w3c/dom/ls/LSInput.java

```diff
@@ -93,15 +95,16 @@
      * using character streams. If an XML declaration is present, the value
      * of the encoding attribute will be ignored.
      */
-    public java.io.Reader getCharacterStream();
+    @Pure
+    public java.io.@Nullable Reader getCharacterStream();
     /**
      *  An attribute of a language and binding dependent type that represents
      * a stream of 16-bit units. The application must encode the stream
      * using UTF-16 (defined in [Unicode] and in [ISO/IEC 10646]). It is not a requirement to have an XML declaration when
      * using character streams. If an XML declaration is present, the value
      * of the encoding attribute will be ignored.
      */
-    public void setCharacterStream(java.io.Reader characterStream);
+    public void setCharacterStream(java.io.@Nullable Reader characterStream);
```

Comment:
@mernst @wmdietl 
I don't think this is the right way to add Annotations?
Is this a bug in AUF?
This java file has numerous of these wrong type to Annotation location.